ui/tests/acceptance/secrets/backend/engines-test.js

```diff
@@ -108,4 +127,122 @@ module('Acceptance | secret-engine list view', function (hooks) {
     await runCmd(deleteEngineCmd(enginePath1));
     await runCmd(deleteEngineCmd(enginePath2));
   });
+
+  test('it allows navigation to a non-nested secret with pagination', async function (assert) {
+    assert.expect(4);
+
+    const enginePath1 = `kv-v1-${this.uid}`;
+    const enginePath2 = `kv-v2-${this.uid}`;
+    const secretPath = 'secret-9';
+
+    await runCmd(mountEngineCmd('kv', enginePath1));
+    await runCmd(mountEngineCmd('kv-v2', enginePath2));
+
+    // check kv1
+    await visit('/vault/secrets');
+    await click(SELECTORS.backendLink(enginePath1));
+    for (let i = 0; i <= 15; i++) {
+      await createSecret(`secret-${i}`, 'foo', 'bar', enginePath1);
+    }
+
+    // navigate and check that details view is shown from non-nested secrets
+    await click(GENERAL.pagination.next);
+    assert.strictEqual(
+      currentURL(),
+      `/vault/secrets/${enginePath1}/list?page=2`,
+      'After clicking next page in navigates to the second page.'
+    );
+    await click(SECRET_ENGINE_SELECTORS.secretLink(secretPath));
+    assert.strictEqual(
+      currentURL(),
+      `/vault/secrets/${enginePath1}/show/${secretPath}`,
+      'After clicking a non-nested secret, it navigates to the details view.'
+    );
+
+    // check kv2 as it follows a different pattern
```

Comment:
I might have steered you wrong here, apologies. There is no need to test KVv2 here as it doesn't use the same route and this test file is just for this route. At least it's delete code rather than add, but all KVv2 things can be deleted, non of the code we touched should impact KVv2.